examples/notebooks/Validating_mechanical_models_Enertech_DFN.ipynb

```diff
@@ -0,0 +1,305 @@
+    "import os\n",
+    "import numpy as np\n",
+    "import matplotlib.pyplot as plt\n",
+    "os.chdir(pybamm.__path__[0]+'/..')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "When you load a model in PyBaMM it builds by default. Building the model sets all of the model variables and sets up any variables which are coupled between different submodels: this is the process which couples the submodels together and allows one submodel to access variables from another. If you would like to swap out a submodel in an exisitng battery model you need to load it without building it by passing the keyword `build=False`"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "model = pybamm.lithium_ion.DFN(\n",
+    "    build=False,options = {\n",
+    "        \"particle\": \"Fickian diffusion\", \n",
+    "        \"cell geometry\": \"arbitrary\", \n",
+    "        \"thermal\": \"lumped\", \n",
+    "        \"particle cracking\": \"no cracking\",\n",
+    "    }\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We can get the parameter set `Ai2020` for the model, which includes the mechanical properties required by the mechanical model."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# param = model.default_parameter_values\n",
+    "chemistry = pybamm.parameter_sets.Ai2020\n",
+    "\n",
+    "# # Below commands are used to add the mechanical parameters missing from other parameter sets \n",
+    "# import pandas as pd\n",
+    "# mechanics = pd.read_csv(\"pybamm/input/parameters/lithium-ion/mechanicals/lico2_graphite_Ai2020/parameters.csv\", \n",
+    "#                        index_col=0, comment=\"#\", skip_blank_lines=True, header=None)[1][1:].dropna().astype(float).to_dict()\n",
+    "# param.update(mechanics, check_already_exists=False)\n",
+    "param = pybamm.ParameterValues(chemistry=chemistry)\n",
+    "experiment05C = pybamm.Experiment([\"Discharge at 0.5C until 3 V\"])\n",
+    "experiment1C = pybamm.Experiment([\"Discharge at 1C until 3 V\"])\n",
+    "experiment2C = pybamm.Experiment([\"Discharge at 2C until 3 V\"])\n",
+    "var = pybamm.standard_spatial_vars\n",
+    "var_pts = {\n",
+    "  var.x_n: 50,\n",
+    "  var.x_s: 50,\n",
+    "  var.x_p: 50,\n",
+    "  var.r_n: 20,\n",
+    "  var.r_p: 20,\n",
+    "}\n",
+    "\n",
+    "experiment = [experiment05C, experiment1C, experiment2C]\n",
+    "solution = []\n",
+    "\n",
+    "for ep in experiment:\n",
+    "    sim = pybamm.Simulation(\n",
+    "        model,\n",
+    "        var_pts=var_pts,\n",
+    "        experiment=ep,\n",
+    "        parameter_values=param,\n",
+    "        solver=pybamm.CasadiSolver(dt_max=0.001)\n",
+    "    )\n",
+    "    solution.append(sim.solve())\n",
+    "\n",
+    "\n",
```

Comment:
Try this instead. It should be much much faster.
```python3
param = pybamm.ParameterValues(chemistry=chemistry)
capacity = param["Cell capacity [A.h]"]
param.update({
    "Current function [A]": capacity * pybamm.InputParameter("C-rate")
})
# experiment05C = pybamm.Experiment(["Discharge at 0.5C until 3 V"])
# experiment1C = pybamm.Experiment(["Discharge at 1C until 3 V"])
# experiment2C = pybamm.Experiment(["Discharge at 2C until 3 V"])
var = pybamm.standard_spatial_vars
var_pts = {
  var.x_n: 50,
  var.x_s: 50,
  var.x_p: 50,
  var.r_n: 20,
  var.r_p: 20,
}

sim = pybamm.Simulation(
        model,
        var_pts=var_pts,
        parameter_values=param,
        solver=pybamm.CasadiSolver(dt_max=600)
    )
Crates = [0.5, 1, 2]
solutions = []

for Crate in Crates:
    print(Crate)
    sol = sim.solve(t_eval=[0, 3600/Crate], inputs={"C-rate": Crate})
    solution.append(sol)


solution05C, solution1C, solution2C = solution
```

Reply:
Is it faster because of not using event or increasing dt_max?

Reply:
It is still using events (the 3V cut-off, which is the value you put in the parameter set called "Lower voltage cut-off [V]").
dt_max is also making it a bit faster.
The main thing is that there is some overhead with creating a simulation, so putting that outside the for loop means it only needs to be done once and then can be reused for each different current. Unfortunately, there is no way to do this right now if using the experiment, but we should definitely work on a way to make that happen.
Also, doing constant-current with an experiment is slightly slower than constant-current without an experiment, because the experiment adds an algebraic equation for the current variable.

Reply:
Is it possible to make this for loop parallel, e.g. [parfor](https://pypi.org/project/parfor/)?

Reply:
@tlestang is working on the parallelization, see #849 to follow progress